buildtest/buildsystem/schemas/utils.py

```diff
@@ -25,9 +25,9 @@ def load_schema(path):
     if not os.path.exists(path):
         sys.exit("schema file %s does not exist." % path)
     with open(path, "r") as fd:
-        if path.endswith("json"):
+        if re.search(".json$", path):
             schema = json.loads(fd.read())
-        elif path.endswith("ml"):
+        elif re.search("(.yaml|.yml)$", path):
             schema = yaml.load(fd.read(), Loader=yaml.SafeLoader)
```

Comment:
This isn't matching what you think - a period means "match anything" For example:

```python
re.search("(.yaml|.yml)$", 'ayaml')                                     
<re.Match object; span=(0, 5), match='ayaml'>
```
What you want is an explicit period, and since it's shared between the two, put it outside of the or:

```python
re.search("[.](yaml|yml)$", 'ayaml')                                    
# No match (correct)
re.search("[.](yaml|yml)$", '.yaml')                                    
<re.Match object; span=(0, 5), match='.yaml'>
re.search("[.](yaml|yml)$", 'config.yaml')                              
<re.Match object; span=(6, 11), match='.yaml'>
```
The same is true for the json above - it needs to be
```python
 re.search("[.]json$", path):
```

Reply:
ahh that's good catch! i totally forgot about the period since it's special character in regex. I will fix this.